lib/ood_core/job/adapters/slurm.rb

```diff
@@ -98,6 +98,25 @@ def initialize(cluster: nil, bin: nil, conf: nil, bin_overrides: {}, submit_host
             @strict_host_checking = strict_host_checking
           end
 
+          # Get a ClusterInfo object containing information about the given cluster
+          # @return [ClusterInfo] object containing cluster details
+          def get_cluster_info
+            def gpus_from_gres(gres)
+              gres.to_s.scan(/gpu:[^,]*(\d+)/).flatten.map(&:to_i).sum
+            end
```

Comment:
I don't think you need this redefinition here anymore.

Reply:
Regex still works on these cases #754. The regex gets the last set of digits before a comma or the end of rthe string if preceded by gpu:, so this holds.

Reply:
> I don't think you need this redefinition here anymore.

gpus_from_gres isn't defined here since it's defined in Slurm while get_cluster_info is defined in Batch, which is nested in Slurm. Should Batch extend Slurm?

Reply:
I see, we can delegate the method to the member variable (`@slurm`) so that it looks as if it's coming from the `Slurm` class.

```ruby
      class Slurm < Adapter
        using Refinements::HashExtensions
        using Refinements::ArrayExtensions

        delegate :gpus_from_gres  to: :slurm
```

It's ruby's synactic sugar for

```ruby 
def gpus_from_gres 
   @slurm.gpus_from_gres 
end
```

Reply:
> I see, we can delegate the method to the member variable (`@slurm`) so that it looks as if it's coming from the `Slurm` class.

The `delegate` method is included in rails, but not ruby. It looks like `def_delegators` from ruby's `forwardable` module should be able to do the same thing, but I can't get that to work either.

Reply:
Lol that's hilarious. This'll do just fine in a pinch.

```ruby
def gpus_from_gres 
   @slurm.gpus_from_gres 
end
```

Reply:
I actually don't think that delegate is right for this. From [rubydocs](https://www.rubydoc.info/docs/rails/4.1.7/Module:delegate) `delegate`:
> Provides a delegate class method to easily expose contained objects' public methods as your own.

So I would want this if I wanted to call a Batch method from Slurm, when I actually need to call a Slurm method from Batch.

Reply:
> This'll do just fine in a pinch.
> 
> ```ruby
> def gpus_from_gres 
>    @slurm.gpus_from_gres 
> end
> ```

That doesn't work either:
```
undefined method `gpus_from_gres' for nil:NilClass
```

Reply:
Right because I've got it backwards. Yea I'd say let's move that function into `Batch` then just reference it through `@slurm.gpus_from_gres`. Other adapters have a helper class which is where it'd go, but alas, we did not use that here.

Reply:
> The regex gets the last set of digits before a comma or the end of the string if preceded by gpu:, so this holds.

But not sure that will work, because the gres won't always be proceeded by `gpu:<digit>`, it might be `gpu:a100:<digit>` because a GRES can have a subtype and users can request sub-types so that would be what shows up.

Reply:
> Right because I've got it backwards. Yea I'd say let's move that function into `Batch` then just reference it through `@slurm.gpus_from_gres`. Other adapters have a helper class which is where it'd go, but alas, we did not use that here.

That doesn't work either. Slurm.parse_job_info can't load `@batch.gpus_from_gres` because it can't access batch. To fix this, I defined `gpus_from_gres` just above the Batch class definition, so that it is accessible everywhere in both classes.